### PR TITLE
Long running tests 814

### LINKED ISSUE #814: Long Running Unit tests (Daily/Weekly?)
Now that we have explored the daily GHA workflow for docs, we should set on up to run the "expensive" marked tests on a regular basis.

### PR DESCRIPTION
**Pull request by j-c-c**
This PR sets up a workflow to run the long running unit tests on our self hosted runners. We have some options as far as the trigger event goes:

EDIT: Based on our discussion in the dev meeting, this workflow will be triggered on push to master of develop.

- ~~Initially we discussed making this a weekly job, but the downside of that is the email notifications on failed jobs are likely to be overlooked.~~
- ~~We can trigger the job on merges. This is accomplished by triggering when a PR has been closed and then running the job on the condition `if_merged`~~
- ~~Another trigger that makes sense is when a PR is "ready for review". This would catch any problems with the long running tests prior to them being merged into develop.~~

This closes #814

### CODE REVIEW

File: .github/workflows/long_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - 'master'
+      - 'develop'
+
+jobs:
+  expensive_tests:
+    runs-on: self-hosted
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: develop
```
Comment (garrettwrong) [inline .github/workflows/long_workflow.yml]: I don't think we want this line anymore

**Review by garrettwrong — CHANGES_REQUESTED** [review]

File: .github/workflows/long_workflow.yml (hunk 1 above)
Comment (j-c-c) [reply .github/workflows/long_workflow.yml]: Oops. Thanks for catching that!

Comment (j-c-c) [reply .github/workflows/long_workflow.yml]: fixed

**Review by garrettwrong — APPROVED** [review]
Awesome. Glad to have this. Now we can start plugging in more tests. Thanks!

**Pull request merged**